component/src/test/java/com/celements/blog/plugin/NewsletterReceiversTest.java

```diff
@@ -19,18 +19,88 @@
+    String email = "test@celements.com";
+    String urlParams = "xpage=celements_ajax&ajax_mode=BlogAjax&doaction=unsubscribe&emailadresse=";
+    String urlStaticPart = "http://celements.com/" + spaceName + "/" + docName + "?" + urlParams;
+    URL url = new URL(urlStaticPart + URLEncoder.encode(email, "UTF-8"));
+    IBlogServiceRole blogServiceMock = registerComponentMock(IBlogServiceRole.class);
+    XWikiDocument doc = new XWikiDocument(new DocumentReference(getContext().getDatabase(),
+        spaceName, docName));
+    BaseObject blogObj = new BaseObject();
+    blogObj.setXClassReference(new DocumentReference(getContext().getDatabase(), "Celements2",
+        "BlogConfigClass"));
+    blogObj.setIntValue("unsubscribe_info", 1);
+    doc.addXObject(blogObj);
+    expect(blogServiceMock.getBlogPageByBlogSpace(eq(spaceName))).andReturn(doc);
+    XWikiURLFactory urlFactoryMock = createMockAndAddToDefault(XWikiURLFactory.class);
+    getContext().setURLFactory(urlFactoryMock);
+    expect(urlFactoryMock.createExternalURL(eq(spaceName), eq(docName), eq("view"), eq(urlParams
+        + URLEncoder.encode(email, "UTF-8")), (String) eq(null), eq(getContext().getDatabase()),
+        same(getContext()))).andReturn(url);
+    replayDefault();
+    String link = comp.getUnsubscribeLink(spaceName, email, getContext());
+    verifyDefault();
+    assertEquals(url.toString(), link);
+  }
+
+  @Test
+  public void testGetUnsubscribeLink_plusInEmail() throws Exception {
+    String spaceName = "TestSpace";
+    String docName = "BlogArticle";
+    String email = "test+test@celements.com";
+    String urlParams = "xpage=celements_ajax&ajax_mode=BlogAjax&doaction=unsubscribe&emailadresse=";
+    String urlStaticPart = "http://celements.com/" + spaceName + "/" + docName + "?" + urlParams;
+    URL url = new URL(urlStaticPart + URLEncoder.encode(email, "UTF-8"));
+    IBlogServiceRole blogServiceMock = registerComponentMock(IBlogServiceRole.class);
+    XWikiDocument doc = new XWikiDocument(new DocumentReference(getContext().getDatabase(),
+        spaceName, docName));
+    BaseObject blogObj = new BaseObject();
+    blogObj.setXClassReference(new DocumentReference(getContext().getDatabase(), "Celements2",
+        "BlogConfigClass"));
+    blogObj.setIntValue("unsubscribe_info", 1);
+    doc.addXObject(blogObj);
+    expect(blogServiceMock.getBlogPageByBlogSpace(eq(spaceName))).andReturn(doc);
+    XWikiURLFactory urlFactoryMock = createMockAndAddToDefault(XWikiURLFactory.class);
+    getContext().setURLFactory(urlFactoryMock);
+    expect(urlFactoryMock.createExternalURL(eq(spaceName), eq(docName), eq("view"), eq(urlParams
+        + URLEncoder.encode(email, "UTF-8")), (String) eq(null), eq(getContext().getDatabase()),
+        same(getContext()))).andReturn(url);
+    replayDefault();
+    String link = comp.getUnsubscribeLink(spaceName, email, getContext());
+    verifyDefault();
+    assertEquals(url.toString(), link);
```

Comment:
the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) also applies to shared testing logic in unit tests, extract method with parameter `String email` and use it the test methods.